client/fingerprint/env_aws.go

```diff
@@ -105,23 +101,19 @@ func (f *EnvAWSFingerprint) Fingerprint(request *FingerprintRequest, response *F
 		"placement/availability-zone": false,
 	}
 	for k, unique := range keys {
-		res, err := client.Get(metadataURL + k)
-		if err != nil {
+		resp, err := ec2meta.GetMetadata(k)
+		if awsErr, ok := err.(awserr.RequestFailure); ok {
```

Comment:
RequestFailures occur when we have a non-200 http code so it's the equivalent to `res.StatusCode != http.StatusOK` part below.